float64 current_x
float64 current_y
---
float64 x
float64 y
float64 distance